frappe/public/js/frappe/form/grid_pagination.js

```diff
@@ -65,7 +112,7 @@ export default class GridPagination {
 
 	get_pagination_html() {
 		let page_text_html = `<div class="page-text">
-				<span class="current-page-number page-number">${__(this.page_index)}</span>
+				<input class="current-page-number page-number" type="text" value="${__(this.page_index)}"/>
```

Comment:
Validation for `min` and `max` values can be added here.

Reply:
I did this
> User can type any number, if the number is less than 1 it changes back to 1, and if the number is more than the total page count it changes back to total page count.

is it not enough?

Reply:
Yes, if we already have validation in JS then it should be enough.